code/game/objects/structures/crates_lockers/closets/secure/security.dm

```diff
@@ -41,20 +41,23 @@
 	desc = "a box with bluespace compression technology that nanotrasen has approved, but this is extremely heavy... If you're glued with this box, pull out of the contents and fold the box."
 	w_class = WEIGHT_CLASS_HUGE
 	drag_slowdown = 4 // do not steal by dragging
-	var/slow_type = MOVESPEED_ID_SLOW_SUITBOX_CAP
 	/* Note for the compression box:
 		Do not put any box (or suit) into this box, or it will allow infinite storage.
 		non-storage items are only legit for this box. (suits are storage too, so, no.)
-		but also, each box should have different 'slow_type' value,
 		nor it will allow a glitch when you can access different boxes at the same time. */
 
 /obj/item/storage/box/suitbox/pickup(mob/user)
 	. = ..()
-	user.add_movespeed_modifier(slow_type, update=TRUE, priority=100, multiplicative_slowdown=4)
+	user.add_movespeed_modifier(MOVESPEED_ID_SLOW_SUITBOX, update=TRUE, priority=100, multiplicative_slowdown=4)
 
-/obj/item/storage/box/suitbox/dropped(mob/user)
+/obj/item/storage/box/suitbox/dropped(mob/living/user)
 	..()
-	user.remove_movespeed_modifier(slow_type, TRUE)
+	var/box_num = 0
+	for(var/obj/item/I in user.get_contents())
+		if(istype(I, /obj/item/storage/box/suitbox))
+			box_num++
+	if(box_num<=0)
+		user.remove_movespeed_modifier(MOVESPEED_ID_SLOW_SUITBOX, TRUE)
```

Comment:
can't you do if (!(type in contents)) and if not just do for (var/obj/item/store/box/suitbox in user contents) return or box_num = TRUE instead of incrementing

Reply:
for(var/.../suitbox in contents)
   remove mod
   break

Reply:
> can't you do if (!(type in contents)) and if not just do for (var/obj/item/store/box/suitbox in user contents) return or box_num = TRUE instead of incrementing

You can't get all types of boxes like that simply

Reply:
(code is moved to `boxes.dm`)
I tried type in contents things, and I don't think it's working in any means. for loop is best.